### PR TITLE
GH-3470 remove legacy SYSTEM repository and related upgrade handling

### PR DESCRIPTION
**Pull request by jeenbroekstra**
GitHub issue resolved: #3470  

Briefly describe the changes proposed in this PR:

- removed legacy SYSTEM Repository code
- cleaned up RepositoryManager API
- updated embedded test servers that still used system repo to create / shut down remote servers
- removed old upgrade handling code that converted legacy SYSTEM repo to new config setup.

----
PR Author Checklist (see the [contributor guidelines](https://github.com/eclipse/rdf4j/blob/main/.github/CONTRIBUTING.md) for more details):

 - [x] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [x] I've added tests for the changes I made
 - [x] I've applied [code formatting](https://github.com/eclipse/rdf4j/blob/main/.github/CONTRIBUTING.md#code-formatting) (you can use `mvn process-resources` to format from the command line)
 - [x] I've [squashed](https://rdf4j.org/documentation/developer/squashing) my commits where necessary 
 - [x] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change

### CODE REVIEW

File: core/repository/manager/src/main/java/org/eclipse/rdf4j/repository/manager/RemoteRepositoryManager.java [hunk 1]
```diff
@@ -218,13 +186,8 @@ public RepositoryConfig getRepositoryConfig(String id) throws RepositoryExceptio
 			protocolSession.setUsernameAndPassword(username, password);
 
 			int serverProtocolVersion = Integer.parseInt(protocolSession.getServerProtocol());
```
Comment (hmottestad) [inline core/repository/manager/src/main/java/org/eclipse/rdf4j/repository/manager/RemoteRepositoryManager.java:188]: Is this variables used anymore? Should we throw an exception if the version is `< 10`?

File: core/repository/manager/src/main/java/org/eclipse/rdf4j/repository/manager/RemoteRepositoryManager.java [hunk 2]
```diff
@@ -233,7 +196,7 @@ public RepositoryConfig getRepositoryConfig(String id) throws RepositoryExceptio
 	}
 
 	@Override
-	public Collection<RepositoryInfo> getAllRepositoryInfos(boolean skipSystemRepo) throws RepositoryException {
```
Comment (hmottestad) [inline core/repository/manager/src/main/java/org/eclipse/rdf4j/repository/manager/RemoteRepositoryManager.java:236]: This isn't backwards compatible and the method wasn't deprecated. Did you introduce a default interface method that is deprecated and maybe logs an error or just simply ignores the provided param?

File: core/repository/manager/src/main/java/org/eclipse/rdf4j/repository/manager/RepositoryManager.java [hunk 3]
```diff
@@ -537,19 +430,29 @@ public RepositoryInfo getRepositoryInfo(String id) throws RepositoryException {
...
 
 	/**
-	 * @param skipSystemRepo
-	 * @throws RepositoryException When not able to retrieve existing configurations
+	 * @deprecated since 4.0 - use {@link #getAllRepositoryInfos()} instead.
 	 */
-	public abstract Collection<RepositoryInfo> getAllRepositoryInfos(boolean skipSystemRepo) throws RepositoryException;
+	@Deprecated
+	public Collection<RepositoryInfo> getAllRepositoryInfos(boolean skipSystemRepo) throws RepositoryException {
+		return getAllRepositoryInfos();
```
Comment (hmottestad) [inline core/repository/manager/src/main/java/org/eclipse/rdf4j/repository/manager/RepositoryManager.java:454]: Yes you did :)

Btw. `@Deprecated` can take arguments now: https://docs.oracle.com/javase/9/core/enhanced-deprecation1.htm#JSCOR-GUID-E182DAF6-282B-4C1F-8CCF-0A61008E3F11

File: site/content/release-notes/4.0.0.md [hunk 4]
```diff
@@ -18,6 +25,20 @@ changes](https://github.com/eclipse/rdf4j/issues?q=is%3Aclosed+is%3Aissue+label%
...
+Older versions of the RDF4J applications can not be upgraded directly to 4.0.0.  This is because support for upgrading from the legacy SYSTEM repository configuration has been removed. We recommend the following upgrade path:
+
+1. upgrade to RDF4J 3.7 and run the application, executing a few queries on one of its repositories. This will upgrade the existing configuration away from the legacy SYSTEM repo used in older application installations.
+2. shut down normally.
+3. do a second upgrade, to RDF4J 4.0.0.
+
+### Removed deprecated initialize methods on Repository, Sail APIs and RepositoryManager
+
+The deprecated `initialize()` methods on `Repository`, `Sail` and `RepositoryManager`  have been removed. Use `init()` instead. (see [GH-1243](https://github.com/eclipse/rdf4j/issues/1243) )
+
```
Comment (hmottestad) [inline site/content/release-notes/4.0.0.md:41]: +1 for updating release notes!

**Review by hmottestad — APPROVED** [review]
Looks good! Had some minor comments but shouldn't hold you up.

**Review by JervenBolleman — APPROVED** [review]
Looks ok to me

**Pull request merged**